simulation/chart.py

```diff
@@ -1,44 +1,147 @@
+    else:
+        return pg.mkColor(color)
+
+
+class PenStyle(Enum):
+    """Pen style for drawing lines"""
+
+    # NoPen                    = QtCore.Qt.PenStyle.NoPen
+    SolidLine = QtCore.Qt.PenStyle.SolidLine  # pyright:ignore [reportGeneralTypeIssues]
+    DashLine = QtCore.Qt.PenStyle.DashLine  # pyright:ignore [reportGeneralTypeIssues]
+    DotLine = QtCore.Qt.PenStyle.DotLine  # pyright:ignore [reportGeneralTypeIssues]
+    DashDotLine = (
+        QtCore.Qt.PenStyle.DashDotLine  # pyright:ignore [reportGeneralTypeIssues]
+    )
+    DashDotDotLine = (
+        QtCore.Qt.PenStyle.DashDotDotLine  # pyright:ignore [reportGeneralTypeIssues]
+    )
+    # CustomDashLine           = QtCore.Qt.PenStyle.CustomDashLine
+    # MPenStyle                = QtCore.Qt.PenStyle.MPenStyle
+
+
+def _make_pen(
+    color: Optional[Color] = None,
+    width: Optional[float] = None,
+    style: Optional[PenStyle] = None,
+) -> Any:
+    """Creates a QPen from a parameters"""
+    config: Dict[str, Any] = {}
+
+    if color is not None:
+        config["color"] = _make_color(color)
+
+    if width is not None:
+        config["width"] = width
+
+    if style is not None:
+        config["style"] = style.value
+
+    return pg.mkPen(**config)
 
-    Args:
-        size (Tuple[int,int]):(width,height) of the charts window.
-        title (str): Title of charts window.
-        show (bool): Show app window.
-        antialias (Optional[bool], optional): Use antialiasing. If true, smooth visuals and slower refresh rate. Defaults to None.
-        foreground (Any, optional): General foreground color (text,ie). Defaults to None.
-    """
-    # Set up PyQtGraph
-    if foreground is not None:
-        pg.setConfigOption("foreground", foreground)
-    if antialias is not None:
-        pg.setConfigOptions(antialias=antialias)
 
-    pg.mkQApp(title)
+class SymbolType(Enum):
+    """Marker symbol shape type"""
 
-    layout = pg.GraphicsLayoutWidget(show=show, title=title)
-    layout.resize(*size)
-    return layout
+    dot = "o"
+    """Circle"""
+    cross = "x"
+    """Cross"""
+    square = "s"
+    """Square"""
 
 
-def create_chart(
+class PenConfig(TypedDict, total=False):
+    """Configuration for pen to draw lines"""
+
+    width: float
+    """Width of the line"""
+    style: PenStyle
+    """Line style"""
+    color: Color
+    """Line color"""
+
+
+class DownsampleMethod(Enum):
+    """Method to resample the data before plotting ot avoid plotting multiple line segments per pixel."""
+
+    subsample = "subsample"
+    """Downsample by taking the first of N samples. This method is fastest and least accurate. """
+    mean = "mean"
+    """Downsample by taking the mean of N samples. """
+    peak = "peak"
+    """Downsample by drawing a saw wave that follows the min and max of the original data. 
+    This method produces the best visual representation of the data but is slower."""
+
+
+def _create_chart(
```

Comment:
Why are these functions outside of `ChartWidget`? If they are static methods, you can use the `@staticmethod` decorator.